### PR TITLE
Wait for damage to activate Gluttony after Neutralizing Gas end

### PR DESCRIPTION
**Pull request by singiamtel**
As suggested by: [Neutralizing Gas & Gluttony](https://github.com/smogon/pokemon-showdown/projects/3#card-52059175)

Code is mostly from [Karthik's fix](https://github.com/smogon/pokemon-showdown/pull/8250), I just fixed the bug that caused the rollback

### CODE REVIEW

File: test/sim/abilities/neutralizinggas.js [hunk 1]
```diff
@@ -201,7 +201,7 @@ describe('Neutralizing Gas', function () {
 
 		// Gluttony now has the opportunity to activate the Aguav Berry again on taking damage
 		battle.makeChoices();
-		assert.equal(wynaut.hp, Math.floor(wynaut.maxhp / 2) - 1 + Math.floor(wynaut.maxhp * 0.33));
+		assert.equal(wynaut.hp, Math.floor(wynaut.maxhp / 2) - 1 + Math.floor(wynaut.maxhp / 3));
```
Comment (singiamtel) [inline test/sim/abilities/neutralizinggas.js:204]: I did have to change the test, was getting off by one on the berry healing

Comment (DaWoblefet) [reply test/sim/abilities/neutralizinggas.js:204]: Looks like this got missed when https://github.com/smogon/pokemon-showdown/commit/05faddbc93f449e6f494e60c5a884dd31a7803fa landed, good catch

**Comment by singiamtel** [comment]
@Karthik99999 could you check the PR? It's mostly the same as #8250

**Review by Karthik99999 — APPROVED** [review]
Looks good to me. Main issue I had was that Gluttony didn't work after a belly drum but the test I wrote for it seems to have passed so I guess it's fine.

**Comment by DaWoblefet** [comment]
Thanks singiamtel!

**Pull request merged**